@here/harp-mapview/lib/MapView.ts

```diff
@@ -3503,15 +3503,16 @@ export class MapView extends THREE.EventDispatcher {
         // running. The initial placement of text in this render call may have changed the loading
         // state of the TextElementsRenderer, so this has to be checked again.
         if (
-            !this.m_firstFrameComplete &&
+            !this.textElementsRenderer.loading &&
+            this.m_visibleTiles.allVisibleTilesLoaded &&
             this.m_initialTextPlacementDone &&
-            !this.isDynamicFrame &&
-            !this.textElementsRenderer.loading
+            !this.m_animatedExtrusionHandler.isAnimating
         ) {
-            this.m_firstFrameComplete = true;
-
-            if (gatherStatistics) {
-                stats.appResults.set("firstFrameComplete", frameStartTime);
+            if (this.m_firstFrameComplete) {
```

Comment:
IMO this should be renamed, as it is no longer the first frame

Reply:
For the performance test we capture the time it took until the first frame is complete. I kept that functionality even though it's questionable why this is done inside MapView and not in the test.